### ISSUE TITLE
Preload's `-p` option seems to be overloaded

### ISSUE THREAD

**Issue by imrehg** [balena-io/balena-cli#1159]
Looking at the help:

```
    --dockerPort, -p <dockerPort>        Docker daemon TCP port number (hint: 2375 for balena devices)
...
    --pin-device-to-release, -p          Pin the preloaded device to the preloaded release on provision    
```
thus `-p` is used twice.... (and just FYI `-P` is already used too.

This is not ideal, and likely prone to error.

**Comment by shaunmulligan** [comment]
nice catch @imrehg . i think we could probably just drop the `-p` from the dockerPort option, so you are always required to specify the full --dockerPort

**Comment by zvin** [comment]
This is a simple change but it will affect all commands that use may use `dockerPort`: preload, deploy and build.

**Comment by pdcastro** [comment]
@shaunmulligan, @zvin, hmm I am not super happy with completely dropping the single-character option for the `dockerPort` option for the `build`, `deploy` and `preload` commands... How about adding `--pin` to mean the same as `--pin-device-to-release`, then drop `-p` for pin? Or least have an alternative short option for either `dockerPort` (build / deploy / preload) or `pin`? E.g. `-i`:

```
balena build -h 192.168.2.3 -p 2375
balena deploy -h 192.168.2.3 -p 2375
balena preload -h 192.168.2.3 -p 2375 -i
balena preload -h 192.168.2.3 -p 2375 --pin
```
Or preserving `-p` for pin:
```
balena build -h 192.168.2.3 -i 2375
balena deploy -h 192.168.2.3 -i 2375
balena preload -h 192.168.2.3 -i 2375 -p
```
By the way, capital `-P` is already taken: `-P <docker> Path to a local docker socket (e.g. /var/run/docker.sock)`. For the options above, my vote would go for preserving `-p` for the build an deploy commands.

**Comment by pdcastro** [comment]
New tentative plan ([flowdock discussion](https://www.flowdock.com/app/rulemotion/r-product/threads/CvnlARvv_5b71EhBBGBqgblAuaH)):
* Drop `-p` for `dockerPort` and keep `-p` for pinning, **but affecting the `balena preload` command only.**
* Keep other commands unaffected. So the `build`, `deploy` and `tunnel` commands all keep `-p` with their current meaning.
* No additional options / aliases (no `--pin`).

This may require some refactoring or copy-and-paste at the code level, but should be “safe” for users (no silent failing of release pinning). Btw, if this work takes place after #1198 is merged, an option would be to convert `balena preload` to oclif instead of refactoring the "Capitano solution".

**Comment by pdcastro** [comment]
Fixed by PR #1201, by removing the `-p` alias for `dockerPort`. Therefore `-p` can only mean `--pin-device-to-release`. This affects the `balena preload` command only.
